calico-enterprise/networking/gateway-api.mdx

```diff
@@ -0,0 +1,325 @@
+- [Use the Gateway API](#use-the-gateway-api)
+- [Disable Gateway API support](#disable-gateway-api-support)
+
+### Enable Gateway API support
+
+To enable Gateway API support, create a `GatewayAPI` resource with name `tigera-secure`:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: operator.tigera.io/v1
+kind: GatewayAPI
+metadata:
+  name: tigera-secure
+EOF
+```
+
+Shortly after this you will see that Gateway API resources are now available:
+
+```bash
+kubectl api-resources | grep gateway.networking.k8s.io
+```
+
+And also that there is a GatewayClass resource corresponding to the Envoy Gateway implementation included in {{prodname}}:
+
+```bash
+kubectl get gatewayclass -o yaml
+```
+
+### Use the Gateway API
+
+Now you can use the Gateway API in any way that the API supports.  When configuring a Gateway resource, be sure to specify `gatewayClassName: tigera-gateway-class`.
+
+By way of a simple example:
+
+- Create echo servers in namespaces `ns1` and `ns2`:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: echoserver
+  namespace: ns1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: echoserver
+  template:
+    metadata:
+      labels:
+        app: echoserver
+    spec:
+      containers:
+      - image: ealen/echo-server:latest
+        imagePullPolicy: IfNotPresent
+        name: echoserver
+        ports:
+        - containerPort: 80
+        env:
+        - name: PORT
+          value: "80"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver
+  namespace: ns1
+spec:
+  ports:
+    - port: 80
+      targetPort: 80
+      protocol: TCP
+  type: ClusterIP
+  selector:
+    app: echoserver
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns2
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: echoserver
+  namespace: ns2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: echoserver
+  template:
+    metadata:
+      labels:
+        app: echoserver
+    spec:
+      containers:
+      - image: ealen/echo-server:latest
+        imagePullPolicy: IfNotPresent
+        name: echoserver
+        ports:
+        - containerPort: 80
+        env:
+        - name: PORT
+          value: "80"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver
+  namespace: ns2
+spec:
+  ports:
+    - port: 80
+      targetPort: 80
+      protocol: TCP
+  type: ClusterIP
+  selector:
+    app: echoserver
+EOF
+```
+
+- Create a client pod that we can test from:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-client
+spec:
+  containers:
+  - args:
+    - "3600"
+    command:
+    - /bin/sleep
+    image: ubuntu
+    imagePullPolicy: Always
+    name: c1
+  terminationGracePeriodSeconds: 0
+EOF
+```
+
+- Once that pod is running, install `curl` in it:
+
+```bash
+kubectl exec -it test-client -- apt-get -y update
```

Comment:
Should we combine the lines and remote `-it`?

Reply:
I don't see a need to do that.  I agree it's slightly annoying that everyone puts `-it` everywhere even when not needed, but it does no harm.  (And I remember cases where `-i` is surprisingly needed.  Not sure if this could be one of those.)

Reply:
I'm confident that these don't need `it`.

Reply:
I think we should use https://hub.docker.com/r/alpine/curl image for this page.
Ubuntu is 400MB+ and requires us to do apt update and install curl, while there is already a 5mb image out there.